sklearn/cross_decomposition/tests/test_pls.py

```diff
@@ -83,25 +83,38 @@ def check_ortho(M, err_msg):
         [[-0.61330704,  0.25616119, -0.74715187],
          [-0.74697144,  0.11930791,  0.65406368],
          [-0.25668686, -0.95924297, -0.11817271]])
-    assert_array_almost_equal(pls_ca.x_weights_, x_weights)
+    x_weights[x_weights == 0] = 1
+    x_weights_quotient = pls_ca.x_weights_ / x_weights
 
     x_rotations = np.array(
         [[-0.61330704,  0.41591889, -0.62297525],
          [-0.74697144,  0.31388326,  0.77368233],
          [-0.25668686, -0.89237972, -0.24121788]])
-    assert_array_almost_equal(pls_ca.x_rotations_, x_rotations)
+    x_rotations[x_rotations == 0] = 1
+    x_rotations_quotient = pls_ca.x_rotations_ / x_rotations
 
     y_weights = np.array(
         [[+0.58989127,  0.7890047,   0.1717553],
          [+0.77134053, -0.61351791,  0.16920272],
          [-0.23887670, -0.03267062,  0.97050016]])
-    assert_array_almost_equal(pls_ca.y_weights_, y_weights)
+    y_weights[x_weights == 0] = 1
+    y_weights_quotient = pls_ca.y_weights_ / y_weights
 
     y_rotations = np.array(
         [[+0.58989127,  0.7168115,  0.30665872],
          [+0.77134053, -0.70791757,  0.19786539],
          [-0.23887670, -0.00343595,  0.94162826]])
-    assert_array_almost_equal(pls_ca.y_rotations_, y_rotations)
+    y_rotations[y_rotations == 0] = 1
+    y_rotations_quotient = pls_ca.y_rotations_ / y_rotations
+
+    # As x_weights = X.dot(x_rotation), sign change should be the same
+    # in both R / python quotient
```

Comment:
We check that sign flipping is the same in x_rotations and x_weights when comparing python to R, as it should be according to mathematical derivation